### ISSUE TITLE
Update Django-rest-framework to latest

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2738]
In a recent PR we were constrained in the update of this library by our then Django 2.2. As from ~~pending~~ now megred PR #2737 we have removed that constraint and cleared the way, in part, for our jump to Django 4.2 LTS.

Follow-up to: "Update Django REST Framework within Django version constraint #2695" #2697

Our current testing branch pyproject.toml has:
> djangorestframework = "==3.13.1"  # Last version to support Django 2.2 (up to 4.0)

PyPi: https://pypi.org/project/djangorestframework/

Project Changelog: https://www.django-rest-framework.org/community/release-notes/#release-notes

Curiously we have there for the latest version:
> Django 4.1 compatibility. [#8591]

However from the GitHub repo README.md 'requirements': https://github.com/encode/django-rest-framework#requirements

> - Python 3.6+
> - Django 4.2, 4.1, 4.0, 3.2, 3.1, 3.0

Presumably pertaining to master branch. So we may want to keep an eye out for 3.14.X versions as we approach Django 4.2.

Regarding the pending 3.15 release, with presumably official Django 4.2, we have the following upstream Milestone:
https://github.com/encode/django-rest-framework/milestone/78

It would seem that we may well be OK to use 3.14 for Django 4.2: from the two comments here in a pending 3.15 Milestone issue: https://github.com/encode/django-rest-framework/pull/8794#issuecomment-1797388221

It seems from comment: https://github.com/encode/django-rest-framework/issues/8942#issuecomment-1542078457
that we can use DRF 3.14 as long as we use at least Django 4.2.1.

**Comment by phillxnet** [comment]
Removing our prior DRF 3.13.1 pinning and we have:
```
lbuildvm:/opt/rockstor # poetry update
Updating dependencies
Resolving dependencies... (1.0s)

Writing lock file

Package operations: 0 installs, 2 updates, 0 removals

  • Updating urllib3 (2.0.7 -> 2.1.0)
  • Updating djangorestframework (3.13.1 -> 3.14.0)
```

So we also have an incidental urllib3 update.
Project Changelog: https://github.com/urllib3/urllib3/blob/main/CHANGES.rst#210-2023-11-13

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2743
